docs/NNNN-direct-clean-path-layout.md

```diff
@@ -0,0 +1,451 @@
+
+The following is an outline to the steps for mapping an identifier/filepath.
+    
+[UTF Replacement Character List](#utf-replacement-character-list)
+is a list of UTF characters mentioned below.
+
+### When `utfEncode` is `true`
+
+1. Replace all non-UTF8 characters with `replacementString`
+2. Split the string at path separator `/`
+3. For each part do the following
+   1. Replace `=` with `=u003D` if it is followed by `u` and four hex digits 
+   2. Replace any character from this list with its utf code in the form `=uXXXX` 
+       where `XXXX` is the code: 
+       `U+0000-U+001F` `U+007F` `U+0020` `U+0085` `U+00A0` `U+1680` `U+2000-U+200F` 
+       `U+2028` `U+2029` `U+202F` `U+205F` `U+3000` `\n` `\t` `*` `?` `:` `[` `]` `"` 
+       `<` `>` `|` `(` `)` `{` `}` `&` `'` `!` `;` `#` `@`  
+   3. If part only contains periods, replace first period (`.`), with UTF Code (`U+002E`)
+   4. Remove part completely, if its length is 0
+   5. When length of part is larger than `maxFilenameLen` use fallback function and return
+4. Join the parts with path separator `/`
+5. When length of result is larger than `maxPathnameLen` use fallback function and return
+
+### When `utfEncode` is `false`
+
+1. Replace all non-UTF8 characters with `replacementString`
+2. Split the string at path separator `/`
+3. For each part do the following
+    1. Replace any whitespace character from this list with `whitespaceReplacementString`: 
+       `U+0009` `U+000A-U+000D` `U+0020` `U+0085` `U+00A0` `U+1680` `U+2000-U+200F`
+       `U+2028` `U+2029` `U+202F` `U+205F` `U+3000`
+    2. Replace any character from this list with `replacementString`: 
+       `U+0000-U+001F` `U+007f` `*` `?` `:` `[` `]` `"`
+       `<` `>` `|` `(` `)` `{` `}` `&` `'` `!` `;` `#` `@`
+    3. Remove leading spaces, `-` and `~` / remove trailing spaces
+    4. If part only contains periods, replace first period (`.`) with `replacementString`
+    5. Remove part completely, if its length is 0
+    6. When length of part is larger than `maxFilenameLen` use fallback function and return
+4. Join the parts with path separator `/`
+5. When length of result is larger than `maxPathnameLen` use fallback function and return
+
+### Fallback function
+
+1. Create digest with `fallbackDigestAlgorithm` from initial parameter as lower case hex string
+2. When digest is longer than `maxFilenameLen` add folder separator `/` after `maxFilenameLen` (character or byte based)
+   until all parts are smaller or equal `maxFilenameLen`
+3. Prepend `fallbackSubdirs` number of digest characters as prefix folders with separator `/`
+4. Prepend `fallbackFolder` with separator `/` 
+
+
+## Examples
+
+### Parameters
+
+It is not necessary to specify any parameters to use the default configuration.
+However, if you were to do so, it would look like the following:
+
+```json
+{
+    "extensionName": "NNNN-direct-clean-path-layout",
+    "maxFilenameLen": 127,
+    "maxFilenameLen": 32000,
+    "utfEncode": false,
```

Comment:
Typo? In the ["Parameter" section above](https://github.com/OCFL/extensions/blob/b98dcdbe1ec2e7006cb122712884a5bfc092fb34/docs/NNNN-direct-clean-path-layout.md#summary), this is defined as `encodeUTF`.

Reply:
yes, everything is encodeUTF now.